### ISSUE TITLE
Wrong number of arguments

### ISSUE THREAD

**Issue by nirname** [rspec/rspec-its#94]
Rails 5.2

```
Using rspec-support 3.12.1
Using rspec-core 3.12.2
Using rspec-expectations 3.12.3
Using rspec-mocks 3.12.6
Using rspec 3.12.0
Using rspec-html-matchers 0.10.0
Using rspec-its 1.3.0
Using rspec-rails 4.0.1
```

When using `[]` for accessing hash-subject attributes

```
Failure/Error: its([:amount]) { is_expected.to eq(1)}

ArgumentError:
  wrong number of arguments (given 2, expected 3..5)
```

https://github.com/rspec/rspec-its/blob/main/lib/rspec/its.rb#L169

**Comment by nirname** [comment]
The part you mention works. It complains about this part https://github.com/rspec/rspec-its/blob/14297a74c9fb6e6501b0a5fb5e3a5b9e4e89bc41/lib/rspec/its.rb#L169
Although these 2 methods should be identical
https://github.com/rspec/rspec-core/blob/1e661db5c5b431c0ee88a383e8e3767f02dccbfe/lib/rspec/core/example_group.rb#L158
and
https://github.com/rspec/rspec-core/blob/1e661db5c5b431c0ee88a383e8e3767f02dccbfe/lib/rspec/core/example_group.rb#L161
vendoring the lib and replacing `example` with `it` helped
Still not sure if it's compatibility problems

**Comment by pirj** [comment]
My apologies for being annoying. Let me repeat my questions.

Can you please provide a bit more context about the spec you’re dealing with?

What if you set the breakpoint [here](https://github.com/rspec/rspec-its/blob/14297a74c9fb6e6501b0a5fb5e3a5b9e4e89bc41/lib/rspec/its.rb#L127), what seems off to you?

**Comment by nirname** [comment]
@pirj Sorry for the long response, I'll try to reproduce this and provide more detailed description

**Comment by paniko0** [comment]
I'm having the same issue. For me, this is happening because the gem `rswag` overrides the method `example` in [here](https://github.com/rswag/rswag/blob/2.15.0/rswag-specs/lib/rswag/specs/example_group_helpers.rb#L93-L113) 

I'm not entirely sure if it's a problem of the `rswag` gem or the `rspec-its` gem. If I were to bet, I'd say changing this gem to call `RSpec::Core::ExampleGroup.example` instead of just `example` in [here](https://github.com/rspec/rspec-its/blob/main/lib/rspec/its.rb#L169) is easier.

**Comment by pirj** [comment]
Do you want to send a PR?

**Comment by paniko0** [comment]
I can certainly push a change, but I can't say this is going to fix OP's issue since OP didn't provide a Gemfile to compare.

**Comment by pirj** [comment]
That’s the best guess we have, and it will fix your issue that we know how to reproduce. If this it turns out not to fix the original issue, the author will reopen with details to reproduce.

**Comment by JonRowe** [comment]
Fixed in #95, I think its bad form to override this method but we have a readily accessible work around in defining our own example creation method and using it.
